src/cheri_sys_regs.sail

```diff
@@ -93,6 +93,30 @@ function legalize_tcc(o : Capability, v : Capability) -> Capability = {
   }
 }
 
+/*
+ * Used during readout (but not assignment!) of ?EPCC registers (CSpecialRW,
+ * handle_trap_extension) and not during control transfer
+ * (prepare_xret_target).
+ *
+ * The result is that it is only possible to faithfully read out ?EPCC if
+ * either
+ *  - misa.C is enabled or
+ *  - misa.C is disabled and the EPCC offset is sufficiently aligned
+ *
+ * If misa.C is held constant, as it almost always will be, between fault and
+ * readback, ?EPCC will be faithfully read back, because the alignment will
+ * adhere to the requirements of the architecture, which is what legalization
+ * is intending to enforce.
+ *
+ * If neither of those conditions hold, the resulting readout will be a lie; if
+ * ?EPCC has additionally been set to be a sentry, then the result will be an
+ * untagged lie.
```

Comment:
I think the conclusion was "don't do that" before. There's a base alignment exception for certain cases but in general if software does stupid things then it can pick up the pieces. You can currently have a misaligned base on MIPS and confuse things that look at PC.